### PR TITLE
String-based config

### PR DESCRIPTION
**Pull request by bygui86**
This is the first part of the config file implementation: string-based.
I will reuse this branch to implement the file-based config later this Friday probably.

### CODE REVIEW

**Comment by bygui86** [comment]
Mathis please add to the README file the explanation of the name `korp`.
I still don't know it :P

File: app/app.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func addGlobalFlags(app *cli.App) {
 			Name:     "config, c",
 			Usage:    "Load configuration from `FILE`",
 			EnvVar:   "KORP_GLOBAL_CONFIG",
-			FilePath: "~/.korp/config",
+			FilePath: "./config",
```
Comment (mkretz) [inline app/app.go]: I don't think the current directory is a good location to read the config from. The user would have to write a config for each directory which they want to work in.

File: patch/action.go [hunk 2]
```diff
@@ -8,16 +8,17 @@ import (
 	"github.com/urfave/cli"
 )
 
-func patch(patchPath, kstPath *string) func(c *cli.Context) error {
+// TODO to be implemented
+// patch - Patch all yaml files in the given path with the new Docker image reference
+func patch(c *cli.Context) error {
```
Comment (mkretz) [inline patch/action.go]: In my opinion (I have changed my mind) we shouldn't implement the patch command in korp. Instead we should document in the the README how the user can use the various options and subcommands of `kustomize` to get what they need.

File: app/app.go [hunk 3]
```diff
@@ -56,40 +59,73 @@ func addGlobalFlags(app *cli.App) {
...
+// setDebug -
+func setDebug(debug bool) {
+
+	if debug {
+		korp_utils.SetLogLevel("debug")
+	}
+}
+
+// loadConfigAsString -
+func loadConfigAsString(wholeConfig string) {
```
Comment (mkretz) [inline app/app.go]: I don't think this is the way to go. I think we should use the pattern suggested by the framework, using [altsrc](https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#without-kubectl-proxy).

File: app/app.go (hunk 1 above)
Comment (bygui86) [reply app/app.go]: Sure, it was just for make easier testing.

File: patch/action.go (hunk 2 above)
Comment (bygui86) [reply patch/action.go]: I agree. We can create a specific PR to completely remove the patch feature.

File: app/app.go (hunk 3 above)
Comment (bygui86) [reply app/app.go]: I need to have a look at that, anyway this was just a workaround waiting for a more elegant solution.

File: patch/action.go (hunk 2 above)
Comment (bygui86) [reply patch/action.go]: Sorry I was following the flow of emails. I approved and already merged the feature branch removing the patch feature.

**Comment by bygui86** [comment]
I think we can close this PR for now, but let's keep the branch in case we need anyway a workaround.

**Pull request closed without merging**